### ISSUE TITLE
Pictures are not send + black preview

### ISSUE THREAD

**Issue by pavelpetrcz** [signalapp/Signal-Android#11244]
Hello,
my wife upgraded to Android 10. Till last update everything was almost always fine. 

Now after selecting media from gallery preview of photo is black. And after attempt to send picture Signal says that sending failed. Status pop up immediatelly.

![obrazek](https://user-images.githubusercontent.com/13877507/116458258-b746d280-a864-11eb-8365-82c5c3ee2214.png)

**Comment by kusal772** [comment]
Please just try by updating the Signal application. It may resolve the problem.

**Comment by UserX404** [comment]
@kusal772
Can you past a link/source where it has been solved?
Nobody here is talking about version. How can you be sure it has been fixed? Please be more specific.
@pavelpetrcz 
What version are you running?
In general: Please don't delete the issue template and append a debug log.

**Comment by pavelpetrcz** [comment]
@kusal772 
App version: 5.10.2
Debug log: https://debuglogs.org/ca78fed3efaf3a567db5e1d4ed99f576a653797a2fde88158ddf9ebac9b41968 
I simulated the situation before upload debug. So it should include needed information. I hope.
If you are able to link this issue with same or similiar issue feel free to close this one as duplicated. 

Thank you, Pavel

**Comment by Kwasow** [comment]
May be connected to #11015 and #9082

**Comment by luken-dev** [comment]
This started to happen on my OP3T after upgrading Android 11 based LineageOS. It seems to happen only with photos taken with OpenCamera with Camera2 API enabled.

If I take a picture with OpenCamera with Original Api, or with LineageOS's built-in camera thumbnails and sharing works fine.

**Comment by phpbg** [comment]
Same bug here, android 11 (lineageOS 18.1), signal 5.10.8, no google services.
* Pictures from camera roll are black and can't be sent.
* Pictures received from other sharing app are visible and can be sent.
* If I open the picture from the file manager, then I can send it with signal and it works

**Comment by phpbg** [comment]
> 
> 
> May be connected to #11015 and #9082

Those reports mention pictures incorrectly displayed (cropped or blurred) and can be workarounded by clearing signal cache.

In this bug report images from camera gallery are just black and clearing signal's cache doesn't help.

**Comment by phpbg** [comment]
Here is my debug log: https://debuglogs.org/34b1dc5e0c8bdf1d4c81e706421db3398e1cced0e692c3bcb72e1239e263a569

I feel those lines may provide useful information:
```
05-17 23:15:24.118 18631 18709 D skia    : libjpeg error 116 <Corrupt JPEG data: 134 extraneous bytes before marker 0xc0> from output_message
05-17 23:15:24.119 18631 18709 D skia    : libjpeg error 116 <Corrupt JPEG data: 134 extraneous bytes before marker 0xc0> from output_message
05-17 23:15:24.119 18631 18709 D skia    : libjpeg error 52 <Quantization table 0x00 was not defined> from output_message
```

But I'm pretty sure all my camera pictures are not corrupted...

**Comment by phpbg** [comment]
@pavelpetrcz Do you use lineageos ?

Just out of curiosity I installed OpenCamera and I can't reproduce the bug in signal with pictures taken from opencamera.

So this may be related to lineageos camera picture format...

**Comment by zutewas** [comment]
@phpbg 
Perhaps there is a relation to my reported issue data yesterday. 

https://github.com/signalapp/Signal-Android/issues/10603

Can u try to delete the exifdata from the pictures which taken with the lineageos camera and try again? In my case i can after this see the picture in gallery and sending works.

**Comment by luken-dev** [comment]
> 
> 
> @pavelpetrcz Do you use lineageos ?
> 
> Just out of curiosity I installed OpenCamera and I can't reproduce the bug in signal with pictures taken from opencamera (using camera2 API).
> 
> So this may be related to lineageos camera picture format...

You could be right claiming that it has more to do with lineageos, rather than signal (at least in my case). The same issue occurs when trying to share the affected images via any other app. I tried emailing it, and on the recipient's end the image is corrupted.

I tried removing exif data with ScrambledExif to no avail - the app shows an error.

Interesting however that 2 gallery apps have no issue rendering said images.

**Comment by pavelpetrcz** [comment]
@luken-dev hi, nope, it is not lineageos, it is regular Android 8/10 on Motorola device. My wife uses preinstalled gallery from OS.

**Comment by zutewas** [comment]
I know that comparing with WhatsApp and Telegram is not correct but both apps have no problem with selecting pictures from own app gallery.

**Comment by cody-signal** [comment]
@pavelpetrcz can you include a recent debug log taken after trying encountering the issue.

I've been trying to repro this with my Pixel 2 running Lineage 18.1 without luck, is there anything special those that are also running lineage are doing to create the media that's failing?

**Comment by zutewas** [comment]
@cody-signal 
Which camera app do you use in the LOS build? 
org.lineageos.snap Version 2.02.072 (b8ff4da1df-00)?
If you try to edit a taken picture with exiftool e.g.

`exiftool.exe "c:\Users\user\Downloads\IMG_20210607_135140.jpg" -Make=cam -o save.jpg` is there an error like this

`Warning: [minor] Entries in IFD0 were out of sequence. Fixed.` ?

**Comment by cody-signal** [comment]
@zutewas I tried with `Version 2.02.072` and `OpenCamera v1.48.1`

**Comment by DarkoVeberic** [comment]
Duplicate of #10394

**Comment by tbart** [comment]
Not a signal user at all, but take a look at the issue I just created:
https://gitlab.com/LineageOS/issues/android/-/issues/4000
I guess this might be a much deeper rabbit hole.
tl;dr:
If anyone is willing to try to issue
```
exiftool -subsectime=123456 broken_image.jpg
```
on a linux box, copy the file back to the phone and report whether that makes the preview thumbnail available for this image (probably clearing cache if signal does some kind of caching with thumbnails?), it'd help to correctly channelize forces for an actual fix (that will have to be in AOSP or ROMs themselves and not signal, IMHO).

Other than that, try to see whether sharing from a file manager (specifically the stock one on lineageos, as that one's tested to work) instead of a gallery or using signal directly works. This would confirm the same nature of the issue as well.

**Comment by birefringence** [comment]
The LOS issue has been fixed in the most recent build.

**Comment by pavelpetrcz** [comment]
wonderful, thans a lot!

**Comment by LeSpocky** [comment]
> The LOS issue has been fixed in the most recent build.

The most recent signal build or the most recent LOS build? 🤔

**Comment by birefringence** [comment]
Sorry for the confusion: In LOS. It's a bug in AOSP 11. The commits referenced here need to be cherry picked from Android 12 to make it work: https://review.lineageos.org/c/LineageOS/android_packages_providers_MediaProvider/+/318282/

**Comment by pavelpetrcz** [comment]
@birefringence @LeSpocky should I reopen the issue and wait for confirmation that fix was successful? or it does not matter?

**Comment by LeSpocky** [comment]
On Tue, Dec 07, 2021 at 11:32:54PM -0800, Pavel Petr wrote:
> @birefringence @LeSpocky should I reopen the issue and wait for confirmation that fix was successful? or it does not matter?

I'm planning to upgrade from LOS 17.1 (Android 10) to 18.1 (Android
11) in the next weeks. I can report. But do as you please, no urgent
need to reopen from my side.

**Comment by luken-dev** [comment]
I'm on LOS 18.1, and can confirm the issue has been fixed. Tested with Camera2API in OpenCamera v1.49.1.

**Comment by phpbg** [comment]
> I'm on LOS 18.1, and can confirm the issue has been fixed. Tested with Camera2API in OpenCamera v1.49.1.

The bug didn't occur with **OpenCamera** anyway. It only happened with **stock camera** app within lineageOS.

Can you test stock camera app and report here the result?

Regards
